packages/redux-subspace-wormhole/src/wormhole.js

```diff
@@ -6,30 +6,33 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import isPlainObject from 'lodash.isplainobject'
+import isPlainObject from "lodash.isplainobject"
+import memoizeOne from "memoize-one"
 
 const wormhole = (mapState, key) => {
+  if (typeof mapState === "string") {
+    const mapStateKey = mapState
+    mapState = state => state[mapStateKey]
 
-    if (typeof mapState === 'string') {
-        const mapStateKey = mapState
-        mapState = (state) => state[mapStateKey]
-
-        if (typeof key === 'undefined') {
-            key = mapStateKey
-        }
+    if (key === undefined) {
+      key = mapStateKey
     }
+  }
 
-    return (store) => ({
-        getState: (next) => () => {
-            const state = next()
+  return store => {
+    const createWormhole = memoizeOne((state, key, value) => ({ [key]: value, ...state }))
 
-            if (isPlainObject(state)) {
-                return { [key]: mapState(store.rootStore.getState()), ...state }
-            } else {
-                return state
-            }
+    return {
+      getState: next => () => {
+        const state = next()
+        if (isPlainObject(state)) {
```

Comment:
I know this was here before, bit when is state not a plain object? Something like immutablejs?

Should the else case give some kind of warning that the store state is not compatible with wormholes?

Reply:
A subspace could map to an array of primative value. In those cases, the state I'd still valid for that subspace, but we cannot apply a wormhole to it so we just don't.

Reply:
Ok fair enough. Seems a bit invisible though.

Out of scope for this but perhaps we should do a one time Dev mode warning on the first encounter of a non object state just to warn the Dev they aren't getting wormholes?

Reply:
I think of rather put effort into completely reworking wormholes to be opt in rather than implicitly available. At that point warning makes more sense as we know the Dev cares. Currently we'd be making a grand assumption that every subspace uses the wormholes which I'm not sure we want to do.